plutus-e2e-tests/plutus-e2e-tests.cabal

```diff
@@ -0,0 +1,153 @@
+cabal-version: 3.0
+name:          plutus-e2e-tests
+version:       1.0.0.0
+license:       Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+
+author:        James Browning
+maintainer:    james.browning@iohk.io
+homepage:      https://github.com/input-output-hk/plutus-apps#readme
+bug-reports:   https://github.com/input-output-hk/plutus-apps/issues
+description:
+  Please see the README on GitHub at <https://github.com/input-output-hk/plutus-apps#readme>
+
+build-type:    Simple
+
+source-repository head
+  type:     git
+  location: https://github.com/input-output-hk/plutus-apps
+
+common lang
+  default-language:   Haskell2010
+  default-extensions:
+    DeriveAnyClass
+    DeriveFoldable
+    DeriveFunctor
+    DeriveGeneric
+    DeriveLift
+    DeriveTraversable
+    ExplicitForAll
+    FlexibleInstances
+    GeneralizedNewtypeDeriving
+    ImportQualifiedPost
+    InstanceSigs
+    MultiParamTypeClasses
+    ScopedTypeVariables
+    StandaloneDeriving
+    TypeFamilies
+
+  ghc-options:
+    -Wall -Widentities -Wincomplete-record-updates
+    -Wincomplete-uni-patterns -Wmissing-import-lists
+    -Wnoncanonical-monad-instances -Wredundant-constraints
+    -Wunused-packages
+
+  -- library
+  --   import:          lang
+  --   hs-source-dirs:  src
+  --   exposed-modules:
+  --     Marconi.CLI
+  --     Marconi.Index.Datum
+  --     Marconi.Index.ScriptTx
+  --     Marconi.Index.Utxo
+  --     Marconi.Indexers
+  --     Marconi.Logging
+  --     Marconi.Orphans
+  --     Marconi.Types
+
+  --------------------
+  -- Local components
+  --------------------
+  --   build-depends:
+  --     , cardano-streaming
+  --     , rewindable-index
+
+  --------------------------
+  -- Other IOG dependencies
+  --------------------------
+  build-depends:
+    , cardano-api
+    , cardano-binary
+    , cardano-ledger-alonzo
+    , cardano-ledger-babbage
+    , cardano-ledger-core
+    , cardano-ledger-shelley
+    , cardano-ledger-shelley-ma
+    , iohk-monitoring
+
+  ------------------------
+  -- Non-IOG dependencies
+  ------------------------
+  build-depends:
+    , base
+    , bytestring
+    , containers
+    , filepath
+    , lens
+    , mwc-random
+    , optparse-applicative
+    , prettyprinter
+    , serialise
+    , sqlite-simple
+    , stm
+    , streaming
+    , text
+    , time
+    , transformers-except
+
+test-suite plutus-e2e-tests-test
+  import:         lang
+  ghc-options:    -Wno-unused-packages
+  type:           exitcode-stdio-1.0
+  main-is:        Spec.hs
+  hs-source-dirs: test
+  other-modules:
+    Helpers
+    PlutusScripts
+    Spec.BabbageFeatures
+    Spec.Builtins.SECP256k1
+
+  --------------------
+  -- Local components
+  --------------------
+  build-depends:  plutus-script-utils
```

Comment:
I recommend to not depend on anything from plutus-apps.

Reply:
Does this mean we cannot define scripts using typed interface?

Reply:
Oh right, just noticed you use `mkUntypedMintingPolicy`. Then, it's okay.

Reply:
I was just thinking that maybe you wanted to keep the list of dependencies as small as possible and only depend on `cardano-api` at most.

Reply:
Can re-implement `Typed` locally to remove this dependency. But at some point it may also make sense to start using `Versioned` to make it easier to run tests in multiple plutus versions, so if it's not a problem I'll keep `plutus-script-utils` as a dependency.